food-truck-frontend/src/components/UserDetails.tsx

```diff
@@ -1,49 +1,96 @@
 import React from 'react';
-import ReviewsList from "./ReviewsList";
-import SubscriptionList from "./SubscriptionList";
-import api from "../util/api";
+import {loadReviewsByUser} from "../api/RateReviewApi";
+import ReviewView from "./truck/rate_and_review/ReviewView";
+import {Grid} from "@material-ui/core";
+import {loadSubscriptionsForUser, loadUserByName} from "../api/UserApi";
+import Review from "../domain/Review";
+import User from "../domain/User";
+import Truck from "../domain/Truck";
+import {List, ListItem} from 'material-ui';
 
 type UserDetailsProps = {
     username: string
 }
 
 type UserDetailsState = {
-    data: {username: string,
-            id: string,
-            email: string}[];
+    user: User | null,
+    reviews: Review[] | null,
+    subscribedTrucks: Truck[] | null,
+    err: string | null
 }
 
 class UserDetails extends React.Component<UserDetailsProps, UserDetailsState>{
     constructor(props: UserDetailsProps) {
         super(props);
-        this.state = {data: [{username: "", id: "", email: ""}]}
+        this.state = {
+            user: null,
+            reviews: null,
+            subscribedTrucks: null,
+            err: null
+        }
     }
 
-    componentDidMount() {
-        api.get(`/search-usernames?username=${this.props.username}`)
-            .then(response => {
-                this.setState({data: response.data});
-            }).catch(error => {
-             console.log(error.toString())
+    async componentDidMount() {
+        const user = await loadUserByName(this.props.username);
+        if (user == null) {
+            this.setState({err: "User not found."});
+            return;
+        }
+
+        const reviewsTask = loadReviewsByUser(user.id, () => {
+            this.setState({err: "Failed to fetch reviews."})
         });
+        const subscriptionsTask = loadSubscriptionsForUser(this.props.username);
+
+        const reviews = await reviewsTask;
+        const subs = await subscriptionsTask;
+
+        if (reviews == null || subs == null) {
+            this.setState({err: "Failed to load user data"});
+        }
+
+        this.setState({user, reviews, subscribedTrucks: subs});
     }
 
     render() {
-        if(this.state.data.length == 0){
-            return(
-                <div>
-                    <h1>User Doesn't Exist</h1>
-                </div>
-            );
-        }
+        if (this.state.err)
+            return <p>{this.state.err}</p>
```

Comment:
If I can get around to it i'll try to update this page to use material-UI stuff but for now it's fine